### ISSUE TITLE
Why no upload?

### ISSUE THREAD

**Issue by duncanphillips** [qstorey/S3Proxy#6]

**Comment by pfdevilliers** [comment]
A possible way to implement this is with go routines. We perform the upload to S3 within a go routine and then return immediately to the caller. We can integrate this with something like SNS to notify the caller when the object has successfully been uploaded OR add in a polling API mechanism to check the progress of the upload.

This really depends on the size of the file being uploaded (or the number of simultaneous uploads).

**Comment by qstorey** [comment]
Yes definitely, uploads should be handled as quickly as possible by storing the content to disk and returning a 200 on the POST request.

By handling HEAD requests (currently not implement), the caller can then make subsequent HEAD requests to determine if the upload to S3 has completed.
